src/MessageManager.cpp

```diff
@@ -207,7 +215,7 @@ void MessageManager::Unsubscribe( IMessageSubscriber* pSubscriber, MessageID m )
 void MessageManager::Broadcast( Message &msg ) const
 {
 	// GAMESTATE is created before MESSAGEMAN, and has several BroadcastOnChangePtr members, so they all broadcast when they're initialized.
-	if(m_Logging)
+	if(this != nullptr && m_Logging)
```

Comment:
Sure, take it out, then go put "local a= nil+1" in some Scripts file.  You can watch the instacrash on startup for yourself.  The best part about your decision to remove the check was that you *left in* the comment explaining why it was there, leaving the code with a comment that makes no sense in context.  Good work.

Putting the nullptr check outside MessageManager::Broadcast is a mistake because it requires more places in the code to know that the problem is possible, introducing more places that can have a mistake or need updated logic later.